status-reporter/status_reporter/rabbitmq_reporter.py

```diff
@@ -13,17 +13,110 @@
 #  * See the License for the specific language governing permissions and
 #  * limitations under the License.
 
-from cloudify.cluster_status import CloudifyNodeType
+import json
 
-from .status_reporter import Reporter
+from requests import get
+from requests.exceptions import RequestException
+
+from cloudify.utils import LocalCommandRunner
+from cloudify.exceptions import CommandExecutionException
+from cloudify.cluster_status import (CloudifyNodeType, ServiceStatus,
+                                     NodeServiceStatus)
+
+from .status_reporter import Reporter, logger
+from .utils import get_systemd_services, get_node_status
+
+RABBITMQ_SERVICE_KEY = 'RabbitMQ'
+CA_PATH = '/etc/cloudify/ssl/rabbitmq-ca.pem'
+RABBITMQ_SERVICES = {'cloudify-rabbitmq.service': 'RabbitMQ'}
+RABBITMQ_URL = 'https://localhost:15671/api/healthchecks/node'
 
 
 class RabbitMQReporter(Reporter):
     def __init__(self):
         super(RabbitMQReporter, self).__init__(CloudifyNodeType.BROKER)
 
     def _collect_status(self):
-        return {}
+        services, statuses = get_systemd_services(RABBITMQ_SERVICES)
+        config = self._config('extra_config', None)
+        if self._rabbitmq_service_not_running(statuses) or not config:
+            return self._rabbitmq_collect_status_failed(services)
+        self._update_node_status(services, statuses, config)
+        self._update_cluster_status(services, statuses, config)
+        status = get_node_status(statuses)
+        return status, services
+
+    @staticmethod
+    def _rabbitmq_service_not_running(statuses):
+        return statuses[0] == NodeServiceStatus.INACTIVE
+
+    def _update_cluster_status(self, services, statuses, config):
+        cluster_status, cluster_extra_info = self._get_cluster_status(config)
+        services[RABBITMQ_SERVICE_KEY]['extra_info']['cluster_status'] = \
+            cluster_extra_info
+        statuses.append(cluster_status)
+
+    def _get_cluster_status(self, config):
```

Comment:
I think this function is redundant, this logic can be in `_rabbitmqctl_cluster_status `